client/src/views/home/HomePage.vue

```diff
@@ -170,8 +177,8 @@ async function login(device: AvailableDevice, password: string): Promise<void> {
   }
 }
 
-function onForgottenPasswordClicked(): void {
-  console.log('forgotten password!');
+function onForgottenPasswordClicked(device: AvailableDevice): void {
+  state.value = HomePageState.ForgottenPassword;
```

Comment:
Set the `selectedDevice` to `device`. In theory this function will only get called by the `LoginPage` and if you're in `LoginPage` it should mean that `selectedDevice` is not null, but as I've said, it's better to keep the states clearly defined and not assume too much.